Runtime/Data/Attribute/RestrictedAttribute.cs

```diff
@@ -0,0 +1,26 @@
+﻿namespace Zinnia.Data.Attribute
+{
+    using UnityEngine;
+    using System;
+
+    /// <summary>
+    /// Defines the <c>[Restricted]</c> attribute.
+    /// </summary>
+    public class RestrictedAttribute : PropertyAttribute
+    {
+        [Flags]
+        public enum Restrictions
```

Comment:
Anything in this class needs docs: Enum, enum's variants, field, constructor. The constructor's summary should be a copy of the type's summary.